libraries/nodejs_install.rb

```diff
@@ -0,0 +1,61 @@
+#
+# Cookbook Name:: languages
+# HWRP:: nodejs_install
+#
+# Copyright 2014, Chef Software, Inc.
+#
+# Licensed under the Apache License, node_version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'open-uri'
+
+class Chef
+  class Resource::NodejsInstall < ChefCompat::Resource
```

Comment:
Like we discussed earlier lets just create separate resource/provider classes by extending from `Resource::LWRPBase`/`Provider::LWRPBase`. It will keep the idioms/flavor of all resources that ship in this cookbook the same. We can update everything at a later point.